data_structures/arrays/get_products_of_all_other_elements.rb

```diff
@@ -85,3 +111,53 @@ def products(nums)
 
 puts(products([1, 2, 3]))
 # => [6, 3, 2]
+
+#
+# Approach 3: O(1) space approach
+#
+
+# This approach is essentially an extension of the approach 2.
+# Basically, we will be using the output array as one of L or R and we will
+# be constructing the other one on the fly.
+
+# Complexity analysis
+#
+# Time complexity: O(N) where N represents the number of elements in the input
+# array. We use one iteration to construct the array L, one to update the array
+# answer.
+
+# Space complexity: O(1) since don't use any additional array for our
+# computations. The problem statement mentions that using the answer
+# array doesn't add to the space complexity.
+
+def products(nums)
+  return [] if nums.size < 2
+
+  res = [1]
+
+  # res[i] contains the product of all the elements to the left
+  # Note: for the element at index '0', there are no elements to the left,
+  # so the res[0] would be 1
+  (0..(nums.size - 2)).each do |i|
+    num = nums[i]
+    res << num * res[i]
+  end
+
+  # product contains the product of all the elements to the right
+  # Note: for the element at index 'length - 1', there are no elements to the right,
+  # so the product would be 1
+  product = 1
+
+  (nums.size - 1).downto(1).each do |i|
```

Comment:
```suggestion
  (nums.size - 1).downto(1) do |i|
```

[`Integer#downto`](https://ruby-doc.org/core-3.0.0/Integer.html#method-i-downto) takes a block and `each` can be omitted.